### ISSUE TITLE
Enhancement: LAN Verifier / Editor - Auto Detect Non-Escaped Apostrophe marks, bad HTML etc

### ISSUE THREAD

**Issue by MikeyGMT** [e107inc/e107#1335]
When populating and verifying a new Lan, if I miss to escape an apostrophe or miss to spot some bad html command, then I get sent to an error page. I can go back and search through each entry.

The idea is to detect the bad line as you exit the field or to highlight before proceeding with the save.

**Comment by CaMer0n** [comment]
:+1:  Thanks. Btw. Are you using the multilan plugin?

**Comment by MikeyGMT** [comment]
I have Multilan, not looked at it recently, I'll download the fresh one. I did not like that it changed admin values e.g. in Links to LAN_ZZZ_123 so you could not see what a link was. If it's the same thing.  

The problem phrase I have found is 
LANINS_144
IMPORTANT: Please copy and paste the contents of the [b]e107.htaccess[/b] into your [b].htaccess[/b] file. Please take care NOT to overwrite any existing data that may be in it.
Welsh 
PWYSIG: Os gwelwch yn dda gopïo a gludo cynnwys y [b]e107.htaccess[/b] i mewn i\'ch [b].htaccess[/b] ffeil. Cymerwch ofal NID i ysgrifennu dros unrhyw ddata sydd eisoes yn bodoli a all fod ynddo.

Bombs every time... ah it does not like the word e107.htaccess 

Hmmm lost about 20 translations :) teach me to save every time.

**Comment by MikeyGMT** [comment]
Oh wow, I'm blocked altogether now :O

**Comment by Moc** [comment]
Quick note regarding `i\'ch`, I'm pretty sure that these escapes are no longer required since the standard is double quotes now. For example:
```php
define("LAN_SOMETHING", "I'm just trying to give an example"); 
```

**Comment by MikeyGMT** [comment]
True, if you are updating the file with an editor, but it bombs when translating on the fly in the application. I've totally locked my home cable IP out of the site. 
Checked IP Deny Manager in cPanel, nothing there to lock me out. Removed the .htaccess. 
Cannot even reach a Test html page in a non-e107 sub-folder. Logging call with my host. 

Can connect via my phone data connection and to e107 as myself.

**Comment by CaMer0n** [comment]
@MikeyGMT Is your IP still whitelisted?

**Comment by MikeyGMT** [comment]
It was, but I have copied the table and deleted the row. 

It wont even load a test page, outside of the e107 fs. htaccess deleted. 

Think it may be the host security, at a level I can't see, even though I have been able to connect to other sites on the same host. 

Anyway, I'm sure the host can help with unlocking my IP. At worst I can hard boot my cable router and probably get an new IP. 

Don't worry about my connection woes :) Would like feedback on the suggestion to verify the lan entries on entry in the verify / edit e107 form?

**Comment by tgtje** [comment]
Like Moc already mentioned >keep it easy example from Dutch (by system or on the fly) 

BELANGRIJK: Kopieër en plak de inhoud van het [b]e107.htaccess[/b] in jouw [b].htaccess[/b] bestand. Wees voorzichtig dat je eventueel bestaande data NIET overschrijft!.

**Comment by MikeyGMT** [comment]
Thanks, of course for the help, as always you guys are great. 
I know it's fast in the file system. By me doing some translating, I hope to improve my skills in php and Welsh. And for e107 I aim to reduce repeated phrases (less work for translators). I hope we can make Translation in the UI faster, easier and more accessible to the non-technical user community, which I hope translates in to more language packs more quickly available :)
